src/main/java/seedu/duke/commands/UpdateCommand.java

```diff
@@ -1,19 +1,155 @@
+import seedu.duke.Parser;
+import seedu.duke.Ui;
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.items.Event;
+import seedu.duke.items.Task;
+
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 
 public class UpdateCommand extends Command {
     protected String listType;
     protected String[] userCommand;
     protected static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy-HHmm");
-
+    protected static final String TITLE_FLAG = "n/";
+    protected static final String DATE_FLAG = "d/";
+    protected static final String VENUE_FLAG = "v/";
+    protected static final String BUDGET_FLAG = "b/";
+    protected static final String DESCRIPTION_FLAG = "p/";
 
     public UpdateCommand(String[] command) {
-
+        this.listType = command[1];
+        this.userCommand = command;
+        ListCommand list = new ListCommand(userCommand);
+        list.execute();
+        System.out.println("Please type the index of the event you would like to update, "
+                + System.lineSeparator() + "and the attribute you would like to update in the format"
+                + System.lineSeparator() + "Index_num [n/TITLE] [d/dd-MM-yyyy HHmm] [p/DESCRIPTION] "
+                + System.lineSeparator() + "Only for Events: [v/VENUE] [b/BUDGET]");
     }
 
     public CommandResult execute() {
-        return new CommandResult("updateCommand");
+        String userInput = Ui.readInput();
+        String[] userUpdates = userInput.split(" ", 10);
+        try {
+            Integer index = Integer.parseInt(userUpdates[0]);
+            if (userUpdates.length == 1) {
+                throw new DukeException("Please specify what to update. ");
+            }
+            for (String update : userUpdates) {
+                if (update.length() >= 2) {
+                    String attribute = update.trim().substring(2);
+                    if (update.contains(TITLE_FLAG)) {
+                        updateTitle(index, userInput);
+                    } else if (update.contains(DATE_FLAG)) {
+                        updateDate(index, userInput);
+                    } else if (update.contains(VENUE_FLAG)) {
+                        updateVenue(index, userInput);
+                    } else if (update.contains(BUDGET_FLAG)) {
+                        updateBudget(index, userInput);
+                    } else if (update.contains(DESCRIPTION_FLAG)) {
+                        updateDescription(index, userInput);
+                    }
+                }
+            }
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        } catch (NullPointerException e) {
+            System.out.println("please follow the error closely");
+        }
+        return new CommandResult("Item Updated");
+    }
+
+
+    protected void updateDate(Integer index, String attribute) {
+        try {
+            if (this.listType.equalsIgnoreCase("-e")) {
```

Comment:
As this looks to be used quite extensively in `UpdateCommand` and possibly other methods/classes in the future, perhaps this could be refactored for neater code i.e. have a boolean method that returns true is `-e` is the `listType`.